server/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java

```diff
@@ -2065,4 +2065,52 @@ public void test_can_write_ignored_object_with_array_column_with_mixed_types() t
         execute("select * from t");
         assertThat(response).hasRows("{items=[42.42, foo]}");
     }
+
+    @Test
+    @UseRandomizedSchema(random = false)
+    public void test_interrupt_remaining_items_with_unassigned_seq_should_not_be_replicated() throws Exception {
+        execute("create table t (a int)");
+
+        Thread insert = new Thread(() -> {
+            try {
+                execute("insert into t select * from generate_series(1, 100000)");
+            } catch (Exception ex) {
+                // JobKilledException but can be also InterruptedException or CancellationException
+                // Ignore, let the test run further
+            }
+        });
+
+        Thread kill = new Thread(() -> {
+            execute("select id from sys.jobs where stmt = 'insert into t select * from generate_series(1, 100000)'");
+            String uuid = (String) response.rows()[0][0];
+            execute("kill ?", new Object[]{uuid});
+        });
+
+
+        insert.start();
+        kill.start(); // without this test passes
+
+        assertBusy(() -> {
+            execute("SELECT underreplicated_shards FROM sys.health WHERE table_name = 't'");
+            assertThat(response).hasRows("0"); // This one fails with KILL
+
+            /*
+            This one doesn't fail even on buggy version when underreplicated_shards >0 and logs are fulls of errors.
+            However, teardown fails with
+            "All incoming requests on node [node_sd2] should have finished. Expected 0 but got 712;"
+
+            execute("optimize table t with (max_num_segments = 1)");
+            execute("""
+                select
+                  sum(translog_stats['uncommitted_operations'])
+                from
+                  sys.shards
+                where
+                  table_name = 't'
+                """
+            );
+            assertThat(response).hasRows("0");
```

Comment:
Not sure why `underreplicated_shards` can verify this and reacts to the fix but `uncommitted_operations` still passes even without the fix (even with logs full of errors).